src/components/shop-dialog.tsx

```diff
@@ -2,22 +2,80 @@
 
 import { faGem, faShoppingCart } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import Image from "next/image";
 import { useRouter } from "next/navigation";
 import { useState } from "react";
 import { api } from "~/api/client";
 import {
   Dialog,
   DialogContent,
-  DialogDescription,
   DialogHeader,
   DialogTitle,
   DialogTrigger,
 } from "~/components/ui/dialog";
 import { getLocale } from "~/i18n";
+import { ItemType } from "~/server/db/client";
+
+function getImageByItemType(itemType: ItemType) {
+  let src;
+  let alt;
+  let amount;
+
+  if (itemType === "food_boost_1000") {
+    src = "food";
+    alt = "food";
+    amount = "x1k";
+  } else if (itemType === "food_boost_1000000") {
+    src = "food";
+    alt = "food";
+    amount = "x1m";
+  } else if (itemType === "wood_boost_1000") {
+    src = "wood";
+    alt = "wood";
+    amount = "x1k";
+  } else if (itemType === "wood_boost_1000000") {
+    src = "wood";
+    alt = "wood";
+    amount = "x1m";
+  } else if (itemType === "gold_boost_1000") {
+    src = "gold";
+    alt = "gold";
+    amount = "x1k";
+  } else if (itemType === "gold_boost_1000000") {
+    src = "gold";
+    alt = "gold";
+    amount = "x1m";
+  } else if (itemType === "stone_boost_1000") {
+    src = "stone";
+    alt = "stone";
+    amount = "x1k";
+  } else if (itemType === "stone_boost_1000000") {
+    src = "stone";
+    alt = "stone";
+    amount = "x1m";
+  }
+
+  return (
+    <div className="relative mr-4 h-24 w-24">
+      <div className="absolute inset-0">
+        <Image
+          width={220}
+          height={220}
+          src={`/resource-images/${src}.png`}
+          alt={alt as string}
```

Comment:
I think TypeScript might even be ok with using the ! operator instead of the parsing to string, which would be less ugly.